File: cookbooks/python/browser-use/schema.py
"""Schema for browser-use cookbook."""

from typing import Optional

from pydantic import BaseModel, Field


class HNPostResult(BaseModel):
    """Structured result saved by each worker to output/result.json."""

    rank: int
    page: int
    position_on_page: int

    title: Optional[str] = None
    hn_item_id: Optional[str] = None
    hn_item_url: Optional[str] = None
    outbound_url: Optional[str] = None
    final_url: Optional[str] = None

    points: Optional[int] = None
    comments: Optional[int] = None

    summary: str  # Required: Markdown summary of the post content
    screenshots: list[str] = Field(default_factory=list)
    actions: list[str] = Field(default_factory=list)
    error: Optional[str] = None

File: cookbooks/typescript/browser-use/schema.ts
/**
 * Schema for browser-use cookbook.
 */

import { z } from "zod";

export const HNPostResultSchema = z.object({
    rank: z.number(),
    page: z.number(),
    position_on_page: z.number(),

    title: z.string().nullable().optional(),
    hn_item_id: z.string().nullable().optional(),
    hn_item_url: z.string().nullable().optional(),
    outbound_url: z.string().nullable().optional(),
    final_url: z.string().nullable().optional(),

    points: z.number().nullable().optional(),
    comments: z.number().nullable().optional(),

    summary: z.string(),
    screenshots: z.array(z.string()).default([]),
    actions: z.array(z.string()).default([]),
    error: z.string().nullable().optional(),
});

export type HNPostResult = z.infer<typeof HNPostResultSchema>;
